File: 0217-contains-duplicate/0217-contains-duplicate.cpp
//optimized approach
//time:o(1)
//space:o(n)
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> s;
        for(int i=0;i<nums.size();i++){
            if(s.find(nums[i]) != s.end()){
                return true;
            }
            s.insert(nums[i]);
        }
        return false;
    }
};


/*
int main(){
    int n;
    cout<<"enter size of array"<<endl;
    cin>>n;
    
    vector<int> nums(n);
    
    cout<<"enter elements"<<endl;
    for(int i=0;i<n;i++){
        cin>>nums[i];
    }
    
    Solution solution;
    
    bool containsDup = solution.containsDuplicate(nums);
    
    if(containsDup){
        cout<<"true"<<endl;
    }
    else{
        cout<<"false"<<endl;
    }
    return 0;
}
*/

File: 0217-contains-duplicate/0217-contains-duplicate.py
##optimized approach
##time:o(1)
##space:o(n)
class Solution:
    def containsDuplicate(self, nums):
        s = set()
        for num in nums:
            if num in s:
                return True
            s.add(num)
        return False




# n = int(input("Enter the size of the array: "))
# nums = []
# print("Enter elements:")
# for _ in range(n):
#     nums.append(int(input()))

# solution = Solution()
# containsDup = solution.containsDuplicate(nums)

# if containsDup:
#     print("true")
# else:
#     print("false")
